src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,172 @@
+            }
+            current = current.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    @Override
+    public void clear() {
+        Arrays.fill(table, null);
+        size = 0;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return size == 0;
+    }
+
+    @Override
+    public Collection<V> values() {
+        List<V> values = new ArrayList<>();
+        for (Node<K, V> node : table) {
+            while (node != null) {
+                values.add(node.value);
+                node = node.next;
+            }
+        }
+        return values;
+    }
+
+    @Override
+    public Collection<K> keySet() {
+        List<K> keys = new ArrayList<>();
+        for (Node<K, V> nodes : table) {
+            Node<K, V> node = nodes;
+            while (node != null) {
+                keys.add(node.key);
+                node = node.next;
+            }
+        }
+        return keys;
+    }
+
+    @Override
+    public boolean containsKey(K key) {
+        for (Node<K, V> node : table) {
+            while (node != null) {
+                if (Objects.equals(node.key, key)) {
+                    return true;
+                }
+                node = node.next;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public boolean containsValue(V value) {
+        for (Node<K, V> node : table) {
+            while (node != null) {
+                if (Objects.equals(node.value, value)) {
+                    return true;
+                }
+                node = node.next;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        sb.append("[ ");
+        for (Node<K, V> nodes : table) {
+            Node<K, V> node = nodes;
+            while (node != null) {
+                sb.append(node.key).append("=").append(node.value).append("  ");
+                node = node.next;
+            }
+        }
+        sb.append("]");
+        return sb.toString();
+    }
+
+    private void resize() {
+        if (size >= threshold) {
+            Node<K, V>[] oldTable = table;
+            table = new Node[oldTable.length * DOUBLE_SIZE];
+            threshold = (int) (table.length * DEFAULT_LOAD_FACTOR);
+            size = 0;
+            for (Node<K, V> nodes : oldTable) {
+                Node<K, V> node = nodes;
+                while (node != null) {
+                    put(node.key, node.value);
+                    node = node.next;
+                }
+            }
+        }
+    }
+
+    private int calculateHash(K hash) {
+        return hash == null ? 0 : Math.abs(hash.hashCode()) % table.length;
+    }
+
+    private static class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(K key, V value, Node<K, V> next) {
```

Comment:
Don't use getters and setters in class Node. Direct access to the fields 'key', 'value', and 'next' should be used instead of public getters and setters since 'Node' is a private inner class.